### DISCUSSION TITLE
A few useful features I would like to request

### DISCUSSION THREAD

**Discussion by Huecuva** [FreeTubeApp/FreeTube#6356] [General]
So having recently started using FreeTube, there are a few features I would love to see added for matters of convenience and usability:

1) A way to reorganize the order of the playlist currently playing while watching a video. 

Current behavior: Playlists can only be reorganized or rearranged in the playlist view where a video cannot be playing while arranging the playlist.

Requested behavior: I often find it convenient and useful to rearrange the play order while a video is playing. Either the current method of editing the playlist order can be implemented in the playlist panel while a video is playing, or a drag and drop functionality similar to how YouTube works can be added. Please correct me if this feature does already exist but I have been unable to locate it.

2) Able to search and browse other channels while a video is playing. 

Current behavior: Searching for videos or browsing a channel will cease all video playback and leave the player screen. 

Requested behavior: YouTube will continue to play a video in a smaller sort of popup window if you scroll down past the video or search for something while the video is playing. It would be great to also see that functionality in FreeTube.

3) Fix screen resizing. 

Current behavior: When attempting to maximize the size of the actual video panel without making FreeTube full screen, there is a delicate balance that must be found between the size of the window limiting the video playback size, and making the window so big that the playlist moves to the side of the video playback and makes it even smaller. 

Requested behavior: Maybe make this functionality a toggle in the settings? Or an option to have the playlist below or to the right?

4) A refresh button at the top near the other navigation buttons. 

Current behavior: I find if I have any given video open long enough, particularly between putting my PC to sleep and waking it back up again, both FreeTube and YouTube will fail to load the video. In FreeTube, I need to either search for the video again or open my playlist view and replay it or click another video in my playlist and then go back to the video in question. There is no way to simply reload the video without leaving the video

Requested behavior: In YouTube I can simply refresh the browser tab and the video will load properly. A refresh or reload button would be much more convenient in FreeTube than the current functionality.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Please search the issue tracker next time:

1. see https://github.com/FreeTubeApp/FreeTube/issues/964
2. https://github.com/FreeTubeApp/FreeTube/issues/5701
3. Sorry but i dont understand what you are requesting. Please explain the issue you are facing and solution
4. see #1005

**Reply by Huecuva** [reply]
1) That's not what I'm referring to. I have noticed that FreeTube marks a video "watched" much too early, and it is kind of annoying, but it's not at all what I'm talking about in my post. What I'm requesting is a way to change the order of the playlist while watching a video and has nothing to do with watched progress. The order of the playlist can currently only be changed in the Playlist view in a sort of edit mode where you can move the video up and down the queue. It would be great to be able to do this while watching a video if, for example, I decide I want to watch a video further down my playlist next instead of the one that is currently queued next in the playlist.

2} Sorry, I thought I did look.

3) When FreeTube is playing a video but the app is not expanded to full screen, the playlist panel can be either below the playback panel if the window itself is small enough, or it snaps to the right of the playback panel if the window is expanded to give it room. When the playlist panel snaps to the right side, it actually shrinks. 

In this screenshot, the FreeTube window is expanded to the maximum size it can be before the Playlist panel snaps to the right, making the playback panel as big as it can possibly be without FreeTube being full screen:
![Perfect Balance](https://github.com/user-attachments/assets/419722ba-cbcb-4cf9-980d-3d916341b7e9)

Here, you will notice that the FreeTube window has been expanded larger than the previous screenshot, but the playback panel is actually smaller because the playlist panel snapped to the right and is taking up more horizontal space:
![Smaller Playback](https://github.com/user-attachments/assets/a1d06c78-9a70-4504-b8db-488443f02554)

4) Sorry. Again, I thought I looked. Coudln't find anything.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
1. Srry not sure how i linked the wrong issue, see #547

**Reply by Huecuva** [reply]
Alright, well apologies for that as well, then. 

Have I made my point 3 as understandable as possible?

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
3. Sorry but i dont see a visual difference in the playlist section if the video is in default or theater mode. Maybe you cam find a difference in pixels and provide those detaila

**Reply by Huecuva** [reply]
There is no actual difference in the playlist panel except that it moves from beneath the playback panel to beside it to the right, forcing the playback panel to shrink, which makes it actually smaller than if the FreeTube window itself was actually small enough to force the Playlist panel to be beneath the playback panel. I haven't actually changed the video mode. I've simply resized the FreeTube window by grabbing the edge of it and dragging it to the right. This creates more horizontal space in the window, allowing the Playlist panel to move up to the right and forcing the video itself to get smaller.  In YouTube, if the browser is not full screen and you resize the window, the Playlist panel does not move from beside the video panel. It will completely disappear if the window is shrunk small enough, but otherwise when the window is expanded, the video panel will expand as expected. At a certain point, the playlist panel will appear and then the video panel will shrink a bit, but it's not as jarring or extreme as when the playlist panel in FreeTube jumps from under the video panel to beside it.

EDIT: Eh...actually I guess it's about the same as the YouTube functionality. I just usually have YouTube in a full screen tab in my browser whereas, as great as FreeTube is, I find it inconvenient to have it full screen so the behavior is more noticeable. Maybe it's just a me thing and it's not actually a problem.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
We are providing the same functionality as YT and this isnt going to change.
